### PR TITLE
Add Feature to Use Config Files

### LINKED ISSUE #4: Use configs file for loading files instead of paths
Set up model paths and parameters with config.yaml instead of using cli.

### PR DESCRIPTION
**Pull request by AjinkyaIndulkar**
This PR adds a new feature to use config file to load pass model checkpoint paths when running dot. 

This feature update maintains backward compatibility with existing CLI options. If a config file is passed, only selective parameters are overriden by the loaded config dictionary.

Closes #4.

#### Added:

- 🔧 added new config files
- ✨ added feature to load config files in `dot.__main__.main()`

#### Updated:

- 📝 updated usage instructions in README

### CODE REVIEW

**Comment by giorgiop** [comment]
Do we want to update also the script without camera, so that they can run using the configs?

**Comment by AjinkyaIndulkar** [comment]
Sure I will update the doc.

**Comment by giorgiop** [comment]
Are we solving this path issues too? https://github.com/sensity-ai/dot/issues/12#issuecomment-1152042735

**Comment by AjinkyaIndulkar** [comment]
> Are we solving this path issues too? [#12 (comment)](https://github.com/sensity-ai/dot/issues/12#issuecomment-1152042735)

Nope. I think it should be handled in a different PR. I believe @Ghassen-Chaabouni might already be working on this.

File: README.md [hunk 1]
```diff
@@ -133,27 +130,22 @@ Run `dot --help` to get a full list of available options.
...
-    --parsing_model_path ./saved_models/simswap/parsing_model/checkpoint/79999_iter.pth \
-    --arcface_model_path ./saved_models/simswap/arcface_model/arcface_checkpoint.tar \
-    --checkpoints_dir ./saved_models/simswap/checkpoints \
-    --crop_size 512 \
     --show_fps \
     --use_gpu
     ```
 
-    Additionally, to enable face superresolution, use the flag `--gpen_type gpen_256` or `--gpen_type gpen_512`.
+    Additionally, to enable face superresolution, change the configuration file to `configs/simswaphq_gpen_256.yaml` or `configs/simswaphq_gpen_512.yaml`.
```
Comment (ghassen1302) [inline README.md:140]: Keep the previous gpen documentation: 

```
use the flag --gpen_type gpen_256 or --gpen_type gpen_512
```

File: configs/simswaphq_gpen_512.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+swap_type: simswap
+parsing_model_path: ./saved_models/simswap/parsing_model/checkpoint/79999_iter.pth
+arcface_model_path: ./saved_models/simswap/arcface_model/arcface_checkpoint.tar
+checkpoints_dir: ./saved_models/simswap/checkpoints
+crop_size: 512
+gpen_type: gpen_512
```
Comment (ghassen1302) [inline configs/simswaphq_gpen_512.yaml:7]: `gpen` can also be used with `simswap` (the 224 model) and with `faceswap_cv2`. I don't think that for every possible combination we should create a config file. We can just add this to the command used `--gpen_type <gpen_type>`

**Review by ghassen1302 — CHANGES_REQUESTED** [review]

File: configs/simswaphq_gpen_512.yaml (hunk 2 above)
Comment (AjinkyaIndulkar) [reply configs/simswaphq_gpen_512.yaml:7]: i see. in that case, I will update the README as well.

File: README.md (hunk 1 above)
Comment (AjinkyaIndulkar) [reply README.md:140]: sure.

btw, all the existing flags are still valid. It's only when you pass a config file path, that certain flags are overriden by the loaded config.

Comment (AjinkyaIndulkar) [reply README.md]: [c89cf77](https://github.com/sensity-ai/dot/pull/17/commits/c89cf77337115f89394187a20e154e63540b7400)

File: configs/simswaphq_gpen_512.yaml (hunk 2 above)
Comment (AjinkyaIndulkar) [reply configs/simswaphq_gpen_512.yaml]: [cf93e9f](https://github.com/sensity-ai/dot/pull/17/commits/cf93e9f6aa2f3445fdb885e231e254720fe9b2cc)

**Review by ghassen1302 — APPROVED** [review]

**Pull request merged**